src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -805,21 +831,27 @@ public String getQuestionResultStatisticsCsv(
         int[][] responseFrequency = RubricStatistics.calculateResponseFrequency(responses, this);
         float[][] rubricStats = RubricStatistics.calculatePercentageFrequencyAndAverage(this,
                 responseFrequency);
+        List<List<Double>> weights = getRubricWeights();
 
         for (int i = 0; i < rubricSubQuestions.size(); i++) {
             String alphabeticalIndex = StringHelper.integerToLowerCaseAlphabeticalIndex(i + 1);
             csv.append(SanitizationHelper.sanitizeForCsv(alphabeticalIndex + ") " + rubricSubQuestions.get(i)));
             boolean isSubQuestionRespondedTo = responseFrequency[i][numOfRubricChoices] > 0;
             for (int j = 0; j < rubricChoices.size(); j++) {
-                String percentageFrequencyString = isSubQuestionRespondedTo
-                                                 ? df.format(rubricStats[i][j] * 100) + "%"
-                                                 : STATISTICS_NO_VALUE_STRING;
-                csv.append("," + percentageFrequencyString + " (" + responseFrequency[i][j] + ")");
+                StringBuilder percentageFrequencyString = new StringBuilder();
+                // Add percentage Frequency, response frequency and if weights are assigned, then add weights.
+                percentageFrequencyString
+                        .append((isSubQuestionRespondedTo
+                                        ? df.format(rubricStats[i][j] * 100) + "%" : STATISTICS_NO_VALUE_STRING)
+                                + " (" + responseFrequency[i][j] + ")"
+                                + (hasAssignedWeights ? " [" + dfWeight.format(weights.get(i).get(j)) + "]" : ""));
```

Comment:
Hmm this chunk of code is exactly same as the one in `getQuestionResultsStatisticsBodyHtml` (lines 728-734). Perhaps we can extract method?

Reply:
Isn't it too small of a chunk to extract this to a separate method of it's own?